### PR TITLE
[init] Initialize Gitlab EE Role

### PR DESCRIPTION
**Pull request by nasirhm**
Fixes half (Configuring Gitlab EE) of #184. 

Initialized Playbook to provision Gitlab EE on a Server.

Tested it locally on CentOS 7.

### CODE REVIEW

**Comment by nazunalika** [comment]
Thank you! I'm going to test it in my lab and see where it goes. I'll comment if there's any issues.

**Comment by nasirhm** [comment]
Awesome, feel free to ping me here or on Slack in case you face any issues with it :)

File: ansible/playbooks/role-gitlab-ee.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+---
+# Creates a Gitlab Instance
+- name: Install and Provision Gitlab
+  hosts: gitlabservers
+  become: true
+
+  pre_tasks:
```
Comment (nazunalika) [inline ansible/playbooks/role-gitlab-ee.yml:7]: You're missing pretasks here... there's an error that comes up about no_ansible not being defined.

```
    - name: Check if ansible cannot be run here
      stat:
        path: /etc/no-ansible
      register: no_ansible
```

File: ansible/playbooks/role-gitlab-ee.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+---
+# Creates a Gitlab Instance
+- name: Install and Provision Gitlab
+  hosts: gitlabservers
+  become: true
+
```
Comment (nazunalika) [inline ansible/playbooks/role-gitlab-ee.yml:6]: Move the vars_files up here if you can

File: ansible/playbooks/role-gitlab-ee.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    - name: Install SELinux packages
+      package:
+        name: python3-policycoreutils.noarch
+        state: present
+
+  roles:
+    - role: geerlingguy.gitlab
+      state: present
+      vars_files:
+        - vars/gitlab.yml
```
Comment (nazunalika) [inline ansible/playbooks/role-gitlab-ee.yml:22]: Move this up to the top, starting at line 6. The vars are not being read and gitlab-ce is being installed instead and a bunch of other variables are ignored.

File: ansible/playbooks/vars/gitlab.yml [hunk 4]
```diff
@@ -0,0 +1,57 @@
+---
+gitlab_domain: src.rockylinux.org
```
Comment (nazunalika) [inline ansible/playbooks/vars/gitlab.yml:2]: Change this to `git.rockylinux.org`

File: ansible/playbooks/vars/gitlab.yml [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+# LDAP Configuration for the future
+gitlab_ldap_enabled: "false"
+gitlab_ldap_host: "example.com"
+gitlab_ldap_port: "389"
+gitlab_ldap_uid: "sAMAccountName"
+gitlab_ldap_method: "plain"
+gitlab_ldap_bind_dn: "CN=Username,CN=Users,DC=example,DC=com"
+gitlab_ldap_password: "password"
+gitlab_ldap_base: "DC=example,DC=com"
```
Comment (nazunalika) [inline ansible/playbooks/vars/gitlab.yml:26]:
```
gitlab_ldap_enabled: "true"
gitlab_ldap_host: "rockylinux.org"
gitlab_ldap_port: "389"
gitlab_ldap_uid: "uid"
gitlab_ldap_method: "start_tls"
gitlab_ldap_bind_dn: "uid=binder,cn=sysaccounts,cn=etc,dc=angelsofclockwork,dc=net"
gitlab_ldap_password: "ThisIsNotThePassword!"
gitlab_ldap_base: "cn=users,cn=accounts,dc=rockylinux,dc=org"
```

File: ansible/playbooks/vars/gitlab.yml [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+gitlab_smtp_port: "465"
+gitlab_smtp_user_name: "smtp user"
+gitlab_smtp_password: "smtp password"
+gitlab_smtp_domain: "example.com"
+gitlab_smtp_authentication: "login"
+gitlab_smtp_enable_starttls_auto: "true"
+gitlab_smtp_tls: "false"
+gitlab_smtp_openssl_verify_mode: "none"
+gitlab_smtp_ca_path: "/etc/ssl/certs"
+gitlab_smtp_ca_file: "/etc/ssl/certs/ca-certificates.crt"
```
Comment (nazunalika) [inline ansible/playbooks/vars/gitlab.yml:51]: Try to change any instance of example.com to `rockylinux.org`

**Comment by nazunalika** [comment]
Hey @nasirhm - So the playbook works! But with some modifications. I've detailed the changes you'll need to make below (in diff format).

Once you've made the changes, I'll take a final look and it'll likely be ready to merge. 👍 Thank you!

You were missing a piece of pre_tasks, and the vars_files needed to be at the top of the file (otherwise gitlab-ce gets installed instead). Here's a diff output.

```
--- /tmp/role.yml       2020-12-17 02:49:31.612280991 +0000
+++ playbooks/role-gitlab-ee.yml        2020-12-17 02:06:58.469460965 +0000
@@ -3,13 +3,21 @@
 - name: Install and Provision Gitlab
   hosts: gitlabservers
   become: true
+  vars_files:
+    - vars/gitlab.yml

   pre_tasks:
+    - name: Check if ansible cannot be run here
+      stat:
+        path: /etc/no-ansible
+      register: no_ansible
+
     - name: Verify if we can run ansible
       assert:
         that:
           - "not no_ansible.stat.exists"
         msg: "/etc/no-ansible exists - skipping un on this node"
+
     - name: Install SELinux packages
       package:
         name: python3-policycoreutils.noarch
@@ -18,8 +26,6 @@
   roles:
     - role: geerlingguy.gitlab
       state: present
-      vars_files:
-        - vars/gitlab.yml

   post_tasks:
     - name: Touching run file that ansible has ran here
```

Some of the variables I had to change up, most of what is there in your PR are the defaults. I changed some of these to reflect what we'll likely see in infrastructure, while maintaining some kind of security (eg with ldap connections)

```
--- /tmp/gitlab.yml     2020-12-17 02:48:58.573103459 +0000
+++ vars/gitlab.yml     2020-12-17 02:52:54.921373469 +0000
@@ -1,5 +1,5 @@
 ---
-gitlab_domain: src.rockylinux.org
+gitlab_domain: git.rockylinux.org
 gitlab_external_url: "https://{gitlab_domain}/"

 # Location where all the git repositories will be stored.
@@ -16,15 +16,14 @@
 gitlab_self_signed_cert_subj: "/C=US/ST=Missouri/L=Saint Louis/O=IT/CN={{ gitlab_domain }}"

 # LDAP Configuration for the future
-gitlab_ldap_enabled: "false"
-gitlab_ldap_host: "example.com"
+gitlab_ldap_enabled: "true"
+gitlab_ldap_host: "rockylinux.org"
 gitlab_ldap_port: "389"
-gitlab_ldap_uid: "sAMAccountName"
-gitlab_ldap_method: "plain"
-gitlab_ldap_bind_dn: "CN=Username,CN=Users,DC=example,DC=com"
-gitlab_ldap_password: "password"
-gitlab_ldap_base: "DC=example,DC=com"
-
+gitlab_ldap_uid: "uid"
+gitlab_ldap_method: "start_tls"
+gitlab_ldap_bind_dn: "uid=binder,cn=sysaccounts,cn=etc,dc=angelsofclockwork,dc=net"
+gitlab_ldap_password: "ThisIsNotThePassword!"
+gitlab_ldap_base: "cn=users,cn=accounts,dc=rockylinux,dc=org"
 gitlab_time_zone: "UTC"

 # Validates Certifications when downloading Gitlab Installation repo
@@ -33,16 +32,16 @@
 # Email and SMTP configuration (For the future)
 # Email configuration.
 gitlab_email_enabled: "false"
-gitlab_email_from: "gitlab@example.com"
+gitlab_email_from: "gitlab@rockylinux.org"
 gitlab_email_display_name: "Gitlab"
-gitlab_email_reply_to: "gitlab@example.com"
+gitlab_email_reply_to: "gitlab@rockylinux.org"
 # SMTP configuration
 gitlab_smtp_enable: "false"
 gitlab_smtp_address: "smtp.server"
 gitlab_smtp_port: "465"
 gitlab_smtp_user_name: "smtp user"
 gitlab_smtp_password: "smtp password"
-gitlab_smtp_domain: "example.com"
+gitlab_smtp_domain: "rockylinux.org"
 gitlab_smtp_authentication: "login"
 gitlab_smtp_enable_starttls_auto: "true"
 gitlab_smtp_tls: "false"
```

I put this info in my reviews too along the code. Thank you again for the PR!

File: ansible/playbooks/vars/gitlab.yml (hunk 6 above)
Comment (nasirhm) [reply ansible/playbooks/vars/gitlab.yml]: https://github.com/rocky-linux/infrastructure/pull/14947/commits/83283fcf4ea9f26af6a80f0cb0aa0797948efc9b

File: ansible/playbooks/vars/gitlab.yml (hunk 5 above)
Comment (nasirhm) [reply ansible/playbooks/vars/gitlab.yml]: Here: https://github.com/rocky-linux/infrastructure/pull/14947/commits/83283fcf4ea9f26af6a80f0cb0aa0797948efc9b

File: ansible/playbooks/vars/gitlab.yml (hunk 4 above)
Comment (nasirhm) [reply ansible/playbooks/vars/gitlab.yml]: here: https://github.com/rocky-linux/infrastructure/pull/14947/commits/83283fcf4ea9f26af6a80f0cb0aa0797948efc9b

File: ansible/playbooks/role-gitlab-ee.yml (hunk 3 above)
Comment (nasirhm) [reply ansible/playbooks/role-gitlab-ee.yml]: Here it's fixed: https://github.com/rocky-linux/infrastructure/pull/14947/commits/83283fcf4ea9f26af6a80f0cb0aa0797948efc9b

Sorry, On the machine that I tested it, I slightly modified it bu add vars_files on top. missed to stage and commit it.

File: ansible/playbooks/role-gitlab-ee.yml (hunk 2 above)
Comment (nasirhm) [reply ansible/playbooks/role-gitlab-ee.yml:8]: https://github.com/rocky-linux/infrastructure/pull/14947/commits/83283fcf4ea9f26af6a80f0cb0aa0797948efc9b

File: ansible/playbooks/role-gitlab-ee.yml (hunk 1 above)
Comment (nasirhm) [reply ansible/playbooks/role-gitlab-ee.yml:9]: https://github.com/rocky-linux/infrastructure/pull/14947/commits/83283fcf4ea9f26af6a80f0cb0aa0797948efc9b

**Comment by nasirhm** [comment]
Thank you very much @nazunalika for the kind review, I've made the changes in the following commit : https://github.com/rocky-linux/infrastructure/pull/14947/commits/83283fcf4ea9f26af6a80f0cb0aa0797948efc9b

**Review by nazunalika — APPROVED** [review]
Looks good to me! The playbooks run. We will get this merged.

**Comment by nasirhm** [comment]
Thank you @nazunalika

**Pull request merged**